### DISCUSSION TITLE
Vibration does not work correctly on ZeppOS 3

### DISCUSSION THREAD

**Discussion by SashaCX75** [zepp-health/zeppos-forum#283] [Developer Feedback]
The first problem is that the name of the vibration types and their descriptions are not true. For example
`vibrate.setMode(VIBRATOR_SCENE_SHORT_LIGHT)`
starts four short vibrations.
And the code 
`vibrate.setMode(VIBRATOR_SCENE_SHORT_MIDDLE)`
starts a vibration of about 1 second. But not 20ms as it is written in the documentation.

The second problem is that, if I understood the description correctly, the code
`vibrate.start(VIBRATOR_SCENE_STRONG_REMINDER)`
should change the type of only one starting vibration. But no matter what type I write, the vibration in this case is always the same.

**Comment by MyoungXUE** [comment]
Hi @SashaCX75 

May I ask what is your device? Because the OTA release time of each device is different, this problem exists on some devices.

The usage of the vibration mode constant will be cancelled later, and numbers will be directly passed in. You can try passing 1-30 to the mode to see if it works.

**Comment by SashaCX75** [comment]
@MyoungXUE 
I tested it on the Amazfit Balance. I think this is the only device on ZeppOS 3 available to users now.
I checked all the variables listed in the documentation. They all work, but they don't match the description.

You write that you can use the numbers 1-30 for the vibration type. Should they correspond to the types specified for API 1, because there too, numbers are used to specify the vibration type, not variables? Besides, there are much less vibration types specified there than 30. I know that originally in the documentation for API 1 more vibration types were listed. But over time some of them were removed from the documentation and I try not to use them in my projects. Because I don't know if they will work fine on all devices or if they were removed from the documentation because there were problems with them.
